### DISCUSSION TITLE
Feedback from migration to UniFiOS 3.x

### DISCUSSION THREAD

**Discussion by pedropombeiro** [unifi-utilities/unifi-common#564] [General]
Given the new features (but also new limitations such as not being able to run podman) in UniFiOS 3.x, I thought it would be interesting to get feedback from users who've taken the leap to upgrade to UniFiOS 3.x, so that users still on 2.x can make an informed decision of when to take the plunge. The idea is that 3.x might contain some features that render some containers used in 2.x redundant. Some examples:

- Is the Multicast Relay container still needed on 3.x, or does UniFi handle SSDP relay across VLANs appropriately nowadays when you have a Chromecast on a different subnet, or even an upstream subnet? Looking at https://www.reddit.com/r/Ubiquiti/comments/137lspi/ssdpmulticast_issues_with_unifi_ap_3rd_party/, I have little hope for it 😬
- Is the Adblocking in UniFiOS 3.x good enough as a Pi-hole replacement for undemanding users?
- Other improvements/limitations you have noticed?

**Comment by Navish360** [comment]
I've been running UniFi OS 3.x across multiple sites without any issues. I still do think Multicast Relay is needed as SSDP across VLAN's seems broken or barely works when it does. I don't think built in Ad-blocking is as robust as a Pi-Hole due to not being able to do custom block list via the GUI. I would say it's definitely the most stable the OS has been so far.... I would recommend moving over to 3.x if you aren't running any custom programs internally on the UDM, UDMSE or UDMP.

**Reply by pedropombeiro** [reply]
Thanks for sharing your experience @Navish360. I gather you're running vanilla 3.x then?

**Comment by emlynmac** [comment]
I've been running 3.0 for a while now, with no issues. I have nspawn container for pi-hole, udm-le and some other custom scripts running. I've had updates apply without breaking and it's been fine.

**Reply by Navish360** [reply]
Interesting. Have you used multicast-relay working on 3,x on the UDMSE? I've been trying to get it to work but most repos aren't updated for 3.x. It would be nice to finally send SSDP over subnets!

**Reply by gatesry** [reply]
@Navish360 - I use a systemd service that fetches the multicast relay script and enables it. See below.

```
[Unit]
Description=Multicast Relay
After=network-online.target

[Service]
ExecStart=/usr/bin/python3 /tmp/multicast-relay.py --interfaces br0 br30
ExecStartPre=/usr/bin/curl -o /tmp/multicast-relay.py \
https://raw.githubusercontent.com/alsmith/multicast-relay/master/multicast-relay.py
Type=oneshot
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

**Reply by Navish360** [reply]
@gatesry do I need to do anything special or just run the script on the UDM-SE?

**Reply by gatesry** [reply]
You have to create a systemd service, which is like a scheduled task in Windows. 

1. In `/etc/systemd/system/` create an empty file called `multicast-relay.service`
2. Input the service contents from my first comment and save it
3. Start the service `systemctl start multicast-relay.service` and enable it on boot `systemctl enable multicast-relay.service`

You can check if it's running using systemctl status
